### PR TITLE
HDDS-10414. Some acceptance tests fail with Docker Compose V2

### PR DESCRIPTION
**Pull request by ArafatKhan2198**
## What changes were proposed in this pull request?
Recon Acceptance test is failing for the robot test :-

```
hadoop-ozone/dist/src/main/smoketest/recon/recon-api.robot
```
The discrepancy arises because, in local Docker setups, DataNodes are named using hyphens, such as `datanode-1`, whereas in the upstream (CI) environment, DataNodes utilize underscores, like `datanode_1`. This variation in naming conventions leads to inconsistencies across different environments.

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-10414
## How was this patch tested?

The acceptance test passed locally :- 
```
------------------------------------------------------------------------------
Check if Recon picks up DN heartbeats                                 | PASS |
------------------------------------------------------------------------------
Check if Recon Web UI is up                                           | PASS |
------------------------------------------------------------------------------
Check web UI access                                                   | PASS |
------------------------------------------------------------------------------
Check admin only api access                                           | PASS |
------------------------------------------------------------------------------
Check unhealthy, (admin) api access                                   | PASS |
------------------------------------------------------------------------------
Check normal api access                                               | PASS |
------------------------------------------------------------------------------
```

And in the forked CI :- https://github.com/ArafatKhan2198/ozone/actions/runs/8030499503

### CODE REVIEW

**Comment by ArafatKhan2198** [comment]
@dombizita @devmadhuu Could you please take a look.

**Comment by adoroszlai** [comment]
@ArafatKhan2198 Please include information about the Docker (and Docker Compose) version where the test is failing.

**Comment by ArafatKhan2198** [comment]
So my local docker versions are the following :- 
```
(base) mohammad.khan:ozone/ (HDDS-10324*) $ docker --version                                    
Docker version 24.0.5, build ced0996
```
```
(base) mohammad.khan:ozone/ (HDDS-10324*) $ docker-compose --version                            
Docker Compose version v2.20.2-desktop.1
```

**Comment by adoroszlai** [comment]
@ArafatKhan2198 please check whether `hadoop-ozone/dist/src/main/smoketest/topology/cli.robot` is also failing with the same Docker version, it has similar host names in assertion:

```
hadoop-ozone/dist/src/main/smoketest/topology/cli.robot
29:                        Should contain   ${output}         10.5.0.7(ozone-topology_datanode_4_1.ozone-topology_net)    IN_SERVICE    /rack2
33:                        Should contain   ${output}         10.5.0.7(ozone-topology_datanode_4_1.ozone-topology_net) IN_SERVICE
36:                        Should contain   ${output}         10.5.0.7(ozone-topology_datanode_4_1.ozone-topology_net)    IN_SERVICE    /rack2
39:                        Should contain   ${output}         10.5.0.7(ozone-topology_datanode_4_1.ozone-topology_net)    IN_SERVICE    /rack2
```

**Comment by ArafatKhan2198** [comment]
Hi @adoroszlai can you please take a look
The forked CI ran green :- https://github.com/ArafatKhan2198/ozone/actions/runs/8229346145

File: hadoop-ozone/dist/src/main/smoketest/topology/cli.robot [hunk 1]
```diff
@@ -22,18 +22,27 @@ Test Timeout        5 minutes
...
 *** Variables ***
 
-
 *** Test Cases ***
 Run printTopology
     ${output} =         Execute          ozone admin printTopology
-                        Should contain   ${output}         10.5.0.7(ozone-topology_datanode_4_1.ozone-topology_net)    IN_SERVICE    /rack2
+                        Should Contain   ${output}         State = HEALTHY
+                        Should Contain          ${output}         IN_SERVICE
+                        Should Match Regexp     ${output}         .*datanode[-_]\\d+.*IN_SERVICE.*
```
Comment (adoroszlai) [inline hadoop-ozone/dist/src/main/smoketest/topology/cli.robot:30]: `State = HEALTHY` was not in the original expected content, why is it added?

Why is check for `rack2` removed?

File: hadoop-ozone/dist/src/main/smoketest/topology/cli.robot [hunk 2]
```diff
@@ -22,18 +22,27 @@ Test Timeout        5 minutes
...
+                        Should Contain   ${output}         State = HEALTHY
+                        Should Contain   ${output}         Location:
+                        Should Match Regexp   ${output}         .*ozone.*datanode[-_]\\d+.*IN_SERVICE.*
+
 Run printTopology --operational-state IN_SERVICE
     ${output} =         Execute          ozone admin printTopology --operational-state IN_SERVICE
-                        Should contain   ${output}         10.5.0.7(ozone-topology_datanode_4_1.ozone-topology_net)    IN_SERVICE    /rack2
+                        Should Contain   ${output}         State = HEALTHY
+                        Should Contain   ${output}         IN_SERVICE
+                        Should Match Regexp   ${output}         .*ozone.*datanode[-_]\\d+.*IN_SERVICE.*
```
Comment (adoroszlai) [inline hadoop-ozone/dist/src/main/smoketest/topology/cli.robot:42]: Let's avoid duplicating assertions.  Please extract a `Keyword` and use it in all test cases.

**Review by adoroszlai — COMMENTED** [review]
Thanks @ArafatKhan2198 for the patch.

File: hadoop-ozone/dist/src/main/smoketest/topology/cli.robot (hunk 1 above)
Comment (ArafatKhan2198) [reply hadoop-ozone/dist/src/main/smoketest/topology/cli.robot]: Each test case begins by checking for the State = HEALTHY output, ensuring that the overall state of the system or the datanodes is as expected.
All the commands start with this line. 

```
sh-4.2$ ozone admin printTopology
State = HEALTHY
 172.19.0.8(ozone-datanode-4.ozone_default):HTTP=9882,CLIENT_RPC=19864,REPLICATION=9886,RATIS=9858,RATIS_ADMIN=9857,RATIS_SERVER=9856,RATIS_DATASTREAM=9855,STANDALONE=9859    IN_SERVICE    /default-rack
 172.19.0.10(ozone-datanode-1.ozone_default):HTTP=9882,CLIENT_RPC=19864,REPLICATION=9886,RATIS=9858,RATIS_ADMIN=9857,RATIS_SERVER=9856,RATIS_DATASTREAM=9855,STANDALONE=9859    IN_SERVICE    /default-rack
 172.19.0.7(ozone-datanode-5.ozone_default):HTTP=9882,CLIENT_RPC=19864,REPLICATION=9886,RATIS=9858,RATIS_ADMIN=9857,RATIS_SERVER=9856,RATIS_DATASTREAM=9855,STANDALONE=9859    IN_SERVICE    /default-rack
```
And I have added the check for rack again

File: hadoop-ozone/dist/src/main/smoketest/topology/cli.robot (hunk 2 above)
Comment (ArafatKhan2198) [reply hadoop-ozone/dist/src/main/smoketest/topology/cli.robot]: Done

**Comment by adoroszlai** [comment]
Thanks @ArafatKhan2198 for updating the patch, latest patch looks good.  Let's have someone with Docker Compose V2 also check it.

**Review by devmadhuu — APPROVED** [review]
Thanks @ArafatKhan2198 for working on the fix. Changes LGTM +1 and tested on following docker and docker-compose versions:
`docker-compose --version`
Docker Compose version v2.18.1
`docker --version`
Docker version 24.0.2, build cb74dfc

```
../test-single.sh om topology/cli.robot 
==============================================================================
Cli :: Smoketest ozone cluster startup                                        
==============================================================================
Run printTopology                                                     | PASS |
------------------------------------------------------------------------------
Run printTopology -o                                                  | PASS |
------------------------------------------------------------------------------
Run printTopology --operational-state IN_SERVICE                      | PASS |
------------------------------------------------------------------------------
Run printTopology --node-state HEALTHY                                | PASS |
------------------------------------------------------------------------------
Cli :: Smoketest ozone cluster startup                                | PASS |
4 tests, 4 passed, 0 failed
==============================================================================

```
```
../test-single.sh om recon/recon-api.robot
==============================================================================
Recon-Api :: Smoke test to start cluster with docker-compose environments.    
==============================================================================
Check if Recon picks up OM data                                       | PASS |
------------------------------------------------------------------------------
Check if Recon picks up DN heartbeats                                 | PASS |
------------------------------------------------------------------------------
Check if Recon Web UI is up                                           | PASS |
------------------------------------------------------------------------------
Check web UI access                                                   | PASS |
------------------------------------------------------------------------------
Check admin only api access                                           | PASS |
------------------------------------------------------------------------------
Check unhealthy, (admin) api access                                   | PASS |
------------------------------------------------------------------------------
Check normal api access                                               | PASS |
------------------------------------------------------------------------------
Recon-Api :: Smoke test to start cluster with docker-compose envir... | PASS |
7 tests, 7 passed, 0 failed
==============================================================================
```

**Review by Tejaskriya — APPROVED** [review]

**Comment by adoroszlai** [comment]
Thanks @ArafatKhan2198 for the patch, @devmadhuu, @Tejaskriya for the review.

**Pull request merged**